src/test/e2e/22_git_and_gitops_test.go

```diff
@@ -50,6 +50,39 @@ func TestGitAndFlux(t *testing.T) {
 	require.NoError(t, err, stdOut, stdErr)
 }
 
+func TestGitAndArgoCD(t *testing.T) {
+	t.Log("E2E: Git and ArgoCD")
+	e2e.SetupWithCluster(t)
+
+	buildPath := filepath.Join("src", "test", "packages", "22-git-and-gitops")
+	stdOut, stdErr, err := e2e.Zarf("package", "create", buildPath, "-o=build", "--confirm", "--skip-sbom")
+	require.NoError(t, err, stdOut, stdErr)
+
+	path := fmt.Sprintf("build/zarf-package-git-data-check-secrets-%s-1.0.0.tar.zst", e2e.Arch)
+	defer e2e.CleanFiles(path)
+
+	// Deploy the gitops example
+	stdOut, stdErr, err = e2e.Zarf("package", "deploy", path, "--confirm")
+	require.NoError(t, err, stdOut, stdErr)
+
+	// This package contains SBOMable things but was created with --skip-sbom
+	require.Contains(t, string(stdErr), "This package does NOT contain an SBOM.")
+
+	tunnel, err := cluster.NewZarfTunnel()
+	require.NoError(t, err)
+	err = tunnel.Connect(cluster.ZarfGit, false)
+	require.NoError(t, err)
+	defer tunnel.Close()
+
+	testGitServerConnect(t, tunnel.HTTPEndpoint())
+	testGitServerReadOnly(t, tunnel.HTTPEndpoint())
+	testGitServerTagAndHash(t, tunnel.HTTPEndpoint())
+	waitArgoDeployment(t)
+
+	stdOut, stdErr, err = e2e.Zarf("package", "remove", "argocd", "--confirm")
+	require.NoError(t, err, stdOut, stdErr)
+}
+
```

Comment:
I don't think this is actually testing argocd (I just see the zarf.yaml at this directory was moved without changes).

Reply:
I think you could potentially call the Argo App and assert on the repoURLS are mutated here as expected

Reply:
This makes sense, however, looking at the Flux test, I don't see anywhere it does anything like this as an example.  I can certainly write a test for this, but calling out that I don't see any obvious place where it tests the mutation of a repository URL via flux.

Reply:
Nevermind, I see it now.

Reply:
I take that back.  It looks like `TestGitAndFlux` is doing two things.  The first part tests the mutation of URLs without flux.  The 2nd part is installing Flux and then waiting for the pod to come online and then tearing it down.  I don't see anywhere where it tests the mutation of the Pod Info Git URL.  Theoretically, it checks to see if the PodInfo pod comes online, which isn't necessarily wrong.  And that is what I was copying for ArgoCD.

I think we could probably split Git, Flux, and ArgoCD into their tests.  Flux needs to have a test that checks the URL mutation (and ArgoCD, the same).  

Separately, on the latest `main`, one of the dependencies is causing immediate test failures.  `https://github.com/defenseunicorns/golang-tekton-hello-world.git`

Reply:
Yeah sorry, I highlighted a bit too far down / should have been clearer - we don't need to repeat the git tests for the argo test - these could/should probably be 3 test functions as you mentioned.

Also agreed checking the GitRepository / Application CRDs would be a good add to both tests if you wanted to do that to tighten up the tests.

Are you sure you have the latest main? That repo was added because it is a single branch, no tag repo that on a full repo clone will cause an already up to date error to be returned from go-git during the fetch after clone (related to this line: https://github.com/defenseunicorns/zarf/blob/main/src/internal/packager/git/clone.go#L66

Reply:
I have decoupled these into 3 separate tests.  I have also added some checks to validate that mutation is working for the "custom resources".